backend/controller/admin/admin_test.go

```diff
@@ -0,0 +1,151 @@
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"connectrpc.com/connect"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	cf "github.com/TBD54566975/ftl/common/configuration"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/alecthomas/assert/v2"
+	"github.com/alecthomas/types/optional"
+)
+
+func TestAdminService(t *testing.T) {
+	config := tempConfigPath(t, "testdata/ftl-project.toml", "admin")
+	ctx := log.ContextWithNewDefaultLogger(context.Background())
+
+	cm, err := cf.New(ctx,
+		cf.ProjectConfigResolver[cf.Configuration]{Config: []string{config}},
+		[]cf.Provider[cf.Configuration]{
+			cf.EnvarProvider[cf.Configuration]{},
+			cf.InlineProvider[cf.Configuration]{},
+		})
+	assert.NoError(t, err)
+
+	// Skip for now, this is failing in GH CI and I give up.
+	// kcp := cf.KeychainProvider{}
```

Comment:
This cannot work outside of local machines, it's designed to use eg. Mac Keychain